tripy/tripy/flat_ir/flat_ir.py

```diff
@@ -59,6 +59,14 @@ def to_mlir_impl():
                 with ir.InsertionPoint(module.body) as ip:
                     # Lets assume only one function with inline code (#9 will fix it)
                     inp_types = [inp.to_mlir() for inp in self.inputs]
+                    if self.shapes:
+                        assert len(inp_types) == len(self.shapes), "mismatched number of inputs vs number of shapes"
```

Comment:
@christopherbate I have a MR for this fix at https://github.com/NVIDIA/TensorRT-Incubator/pull/43/. It is done in a different manner such that shape is passed on to internal IR tensors as well (when its known).